### PR TITLE
[Feat] (1) GMP API suite fix (2) Create equality & UT for equality, (3) Refactor tests file structure

### LINKED ISSUE #11: [Feature Request] API for converting GMP large number to S21_DECIMAL
Function `convert_gmp_to_decimal` is required to

**Tasks:**
- [x] Convert all  gmp integers to decimal (find way to get individual bits of GMP)
- [x] Get fractional part from gmp floats (see GMP docs & find special function)
- [x] Get rid of the fractional part by multiplying gmp float by 10
- [x] Get exponent from GMP float (this is the number of time we multiplied by 10)
- [x] Convert GMP integer to decimal (see step 1)

👨‍💻 [GMP docs](https://gmplib.org/manual/)

**Bonus:**
- [x] Write unit tests for unit testing API using different tool, i.e. GCC __int128

### PR DESCRIPTION
**Pull request by posidoni**
🔋 **(1) GMP API suite fix**
- GMP <--> s21_decimal API suite is fixed. The problem was in import/export fns.
- Created stable UT's for this API
 
⚖️  **(2) Create equality & UT for equality**
 - Refactored equality, created UT's for this module, passed all tests (stable)
 
 📂 **(3) Refactor tests file structure**
 - Refactored tests API, made some functions private

### CODE REVIEW

**Comment by posidoni** [comment]
Removed ASAN from Makefile, as it failed Alpine build.

**Comment by bezlant** [comment]
Equality I LIKE [🏳️‍🌈](https://emojipedia.org/pride/)

**Review by bezlant — APPROVED** [review]
Good stuff. Appreciate the cleaning up the code & restructuring it.

File: src/tests/helpers/random_values.c [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                ADD_BIT(in->bits[j], i);
+                mpz_bin_str[m] = '1';
+            } else {
+                ZERO_BIT(in->bits[j], i);
+                mpz_bin_str[m] = '0';
+            }
+        }
+    }
+
+    /* This reverse is needed to match the format of normal binary with subject */
```
Comment (bezlant) [inline src/tests/helpers/random_values.c:42]: Might need to remove this and restructure all mpz's to big endian & convert it to whatever endian once we are ready to give results back.

**Review by bezlant — APPROVED** [review]

**Pull request merged**